bundles/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/message/PowermaxBaseMessage.java

```diff
@@ -147,17 +164,55 @@ public boolean isAckRequired() {
         return receiveType == null || receiveType.isAckRequired();
     }
 
+    // Debugging helpers
+
+    public void addDebugInfo(String name, String info, String decoded) {
+        StringBuilder debugLine = new StringBuilder();
+
+        debugLine.append(name);
+        debugLine.append(" = ");
+        debugLine.append(info);
+
+        if ((decoded != null) && !(decoded.isBlank())) {
+            debugLine.append(" - " + decoded);
+        }
+
+        debugInfo.add(debugLine.toString());
+    }
+
+    public void addDebugInfo(String name, String info) {
+        addDebugInfo(name, info, null);
+    }
+
+    public void addDebugInfo(String name, byte[] data, String decoded) {
+        String hex = "0x" + HexUtils.bytesToHex(data);
+        addDebugInfo(name, hex, decoded);
+    }
+
+    public void addDebugInfo(String name, byte[] data) {
+        addDebugInfo(name, data, null);
+    }
+
+    public void addDebugInfo(String name, int data, String decoded) {
+        String len = data <= 0xFF ? "02" : data <= 0xFFFF ? "04" : "08";
+        String hex = String.format("0x%" + len + "X", data);
+        addDebugInfo(name, hex, decoded);
+    }
+
+    public void addDebugInfo(String name, int data) {
+        addDebugInfo(name, data, null);
+    }
+
```

Comment:
You can call `addDebugInfo(String, byte)` and it will happily promote the byte to an int and use this variant. This is already being done in some places and it seems to work fine. I didn't add a byte variant to avoid unnecessary duplication, but I can if you think it's important?

Reply:
Is it called sometimes with an int or is it always with a byte as parameter ?

Reply:
Yes, in PowermaxStatusMessage for example:

```
        byte[] message = getRawData();
        byte eventType = message[3];
        String eventTypeStr = PowermaxMessageConstants.getMessageTypeString(eventType & 0x000000FF);

        addDebugInfo("Event type", eventType, eventTypeStr);
```